py/server/zen_balancer.py

```diff
@@ -0,0 +1,199 @@
+                    }
+                }
+
+        print_outgoing("BalancerApiServer", "/api/v1/getOwnerships", ret)
+
+        return json.dumps(ret)
+
+    @app.route('/api/v1/getOwnerScAddresses', methods=['POST'])
+    def get_owner_sc_addresses():
+        input = json.loads(request.data)
+
+        print_incoming("BalancerApiServer", "/api/v1/getOwnerScAddresses", input)
+
+        if mock_nsc:
+            ret = MOCK_OWNER_SC_ADDR_LIST
+        else:
+            try:
+                ret = get_owner_sc_addr_list()
+            except Exception as e:
+                ret = {
+                    "error": {
+                        "code": 302,
+                        "description": "Could not get owner sc addresses",
+                        "detail": "An exception occurred: " + str(e)
+                    }
+                }
+
+        print_outgoing("BalancerApiServer", "/api/v1/getOwnerScAddresses", ret)
+
+        return json.dumps(ret)
+
+    @app.route('/api/v1/getProposals', methods=['POST'])
+    def get_proposals():
+        input = json.loads(request.data)
+
+        print_incoming("BalancerApiServer", "/api/v1/getProposals", input)
+        response = [prop.to_json() for prop in proposal_dict.values()]
+        print_outgoing("BalancerApiServer", "/api/v1/getProposals", response)
+
+        return json.dumps(response)
+
+    @app.route('/api/v1/createProposal', methods=['POST'])
+    def create_proposal():
+
+        proposal = json.loads(request.data)
+        print_incoming("BalancerApiServer", "/api/v1/createProposal", proposal)
+        try:
+            # update MC chain tip. This block will be used when retrieving balances from Rosetta
+            (chain_tip_height, chain_tip_hash) = get_chain_tip()
+        except Exception as e:
+            response = {
+                "error": {
+                    "code": 303,
+                    "description": "Can not create proposal",
+                    "detail": "can not determine main chain best block: " + str(e)
+                }
+            }
+        else:
+            try:
+                store_proposal_data(proposal, chain_tip_height, chain_tip_hash)
+            except Exception as e:
+                response = {
+                    "error": {
+                        "code": 304,
+                        "description": "Can not create proposal",
+                        "detail": "proposal data format not expected: " + str(e)
+                    }
+                }
+            else:
+                response = {"status": "Ok"}
+
+        print_outgoing("BalancerApiServer", "/api/v1/createProposal", response)
+        return json.dumps(response)
+
+    @app.route('/api/v1/getVotingPower', methods=['GET'])
+    def get_voting_power():
+
+        content = request.args
+        print_incoming("BalancerApiServer", "/api/v1/getVotingPower", content)
+
+        if get_active_proposal().is_null():
```

Comment:
Since there is already a check for active proposal in `get_address_balance` in `rosetta_methods.py`, is it necessary to check it here as well?

Reply:
Done